From: Ervin Hegedus <airween@gmail.com>
Date: Fri, 8 Feb 2019 13:12:04 +0100
Subject: Avoid to use ENV:TERM in the regression tests

There was a bug in Debian build flows: the sbuild environment doesn't
sets the TERM environment, which was used until the tests.
This issue had been fixed in upstream, but will release just in
next release.
---
 src/actions/set_env.cc                       |  2 +-
 test/regression/regression.cc                |  5 +++
 test/test-cases/regression/variable-ENV.json | 55 +++++++++++++++++++++++++---
 3 files changed, 56 insertions(+), 6 deletions(-)

diff --git a/src/actions/set_env.cc b/src/actions/set_env.cc
index 82381ff..ad54170 100644
--- a/src/actions/set_env.cc
+++ b/src/actions/set_env.cc
@@ -37,7 +37,7 @@ bool SetENV::evaluate(Rule *rule, Transaction *t) {
     ms_dbg_a(t, 8, "Setting envoriment variable: "
         + colNameExpanded + ".");
 
-    putenv((char *)colNameExpanded.c_str());
+    putenv(strdup(colNameExpanded.c_str()));
 
     return true;
 }
diff --git a/test/regression/regression.cc b/test/regression/regression.cc
index 15721a1..6f5b3a4 100644
--- a/test/regression/regression.cc
+++ b/test/regression/regression.cc
@@ -424,6 +424,11 @@ after_debug_log:
 
 int main(int argc, char **argv) {
     ModSecurityTest<RegressionTest> test;
+
+    std::string ver(MODSECURITY_VERSION);
+    std::string envvar("MODSECURITY=ModSecurity " + ver + " regression tests");
+
+    putenv(strdup(envvar.c_str()));
 #ifndef NO_LOGS
     int test_number = 0;
 #endif
diff --git a/test/test-cases/regression/variable-ENV.json b/test/test-cases/regression/variable-ENV.json
index b5d0c70..c890b95 100644
--- a/test/test-cases/regression/variable-ENV.json
+++ b/test/test-cases/regression/variable-ENV.json
@@ -2,7 +2,7 @@
   {
     "enabled":1,
     "version_min":300000,
-    "title":"Testing Variables :: ENV (1/3)",
+    "title":"Testing Variables :: ENV (1/4)",
     "client":{
       "ip":"200.249.12.31",
       "port":123
@@ -46,7 +46,7 @@
   {
     "enabled":1,
     "version_min":300000,
-    "title":"Testing Variables :: ENV (2/3)",
+    "title":"Testing Variables :: ENV (2/4)",
     "client":{
       "ip":"200.249.12.31",
       "port":123
@@ -80,17 +80,62 @@
       ]
     },
     "expected":{
-      "debug_log":"Variable: ENV:TERM"
+      "debug_log":"regression tests\" .Variable: ENV:MODSECURITY."
     },
     "rules":[
       "SecRuleEngine On",
-      "SecRule ENV:TERM \"@contains test\" \"id:1,phase:3,pass,t:trim\""
+      "SecRule ENV:MODSECURITY \"@contains test\" \"id:1,phase:3,pass,t:trim\""
     ]
   },
   {
     "enabled":1,
     "version_min":300000,
-    "title":"Testing Variables :: ENV (3/3)",
+    "title":"Testing Variables :: ENV (3/4)",
+    "client":{
+      "ip":"200.249.12.31",
+      "port":123
+    },
+    "server":{
+      "ip":"200.249.12.31",
+      "port":80
+    },
+    "request":{
+      "headers":{
+        "Host":"localhost",
+        "User-Agent":"curl/7.38.0",
+        "Accept":"*/*",
+        "Content-Length": "27",
+        "Content-Type": "application/x-www-form-urlencoded"
+      },
+      "uri":"/",
+      "method":"POST",
+      "body": [
+        "param1=value1&param2=value2"
+      ]
+    },
+    "response":{
+      "headers":{
+        "Date":"Mon, 13 Jul 2015 20:02:41 GMT",
+        "Last-Modified":"Sun, 26 Oct 2014 22:33:37 GMT",
+        "Content-Type":"text/html"
+      },
+      "body":[
+        "no need."
+      ]
+    },
+    "expected":{
+      "debug_log":"Target value: \"bar\" .Variable: ENV:modfoo."
+    },
+    "rules":[
+      "SecRuleEngine On",
+      "SecRule ARGS \"@contains value\" \"id:1,phase:3,pass,t:trim,setenv:modfoo=bar\"",
+      "SecRule ENV:modfoo \"@contains test\" \"id:2,phase:3,pass,t:trim\""
+    ]
+  },
+  {
+    "enabled":1,
+    "version_min":300000,
+    "title":"Testing Variables :: ENV (4/4)",
     "client":{
       "ip":"200.249.12.31",
       "port":123
